src/main/java/org/casbin/shiro/authorize/rbac/program/Operation.java

```diff
@@ -28,20 +30,114 @@
  */
 public class Operation {
 
+    private final Enforcer e;
+
+    public Operation(Enforcer e) {
+        this.e = e;
+    }
+
+    /**
+     * hasAuthorization determines whether the user currently logged on has a permission of the object.
+     *
+     * @param act the action (usually means the operation that the user performs on the resource).
+     * @return whether the current login user has the permission.
+     */
+    public boolean hasAuthorization(String act) {
+        if (isLogin()) {
+            return false;
+        }
+        Subject subject = SecurityUtils.getSubject();
+        return e.enforce(subject.getPrincipal().toString(), act);
+    }
+
     /**
-     * hasAuthorization determines whether the current login user has a permission.
+     * hasAuthorizations determines whether the user currently logged on has a permission of the object.
      *
-     * @param e the enforcer;
      * @param obj the object (usually means resources).
      * @param act the action (usually means the operation that the user performs on the resource).
      * @return whether the current login user has the permission.
      */
-    public static boolean hasAuthorization(Enforcer e, String obj, String act) {
-        if (!SecurityUtils.getSubject().isAuthenticated()) {
-            LogUtil.logPrintfError("Please login");
+    public boolean hasAuthorization(String obj, String act) {
+        if (isLogin()) {
             return false;
         }
         Subject subject = SecurityUtils.getSubject();
         return e.enforce(subject.getPrincipal().toString(), obj, act);
     }
+
+    /**
+     * hasAuthorizations determines whether the user currently logged on has a permission of the object.
+     *
+     * @param dom the domain the roles belong to.
+     * @param obj the object (usually means resources).
+     * @param act the action (usually means the operation that the user performs on the resource).
+     * @return whether the current login user has the permission.
+     */
+    public boolean hasAuthorization(String dom, String obj, String act) {
```

Comment:
`hasAuthorization` is not correct English. Should be `isAuthorized` or `isPermitted` or `hasPermission`.

Reply:
Ok,I will take your advice.